kazoo/protocol/connection.py

```diff
@@ -439,7 +439,7 @@ def _send_request(self, read_timeout, connect_timeout):
         if request.type == Auth.type:
             xid = AUTH_XID
         else:
-            self._xid += 1
+            self._xid = (self._xid % 2147483647) + 1
```

Comment:
kazoo/protocol/connection.py, line 276
b.extend(int_struct.pack(xid))

xid += 1 in python can be bigger than 2147483648, but int_struct.pack(xid) just pack xid to int of 4 bytes, so xid must  in range -2147483648~2147483647